src/Build/Evaluation/SimpleProjectRootElementCache.cs

```diff
@@ -63,8 +63,11 @@ private ProjectRootElement GetFromOrAddToCache(string projectFile, OpenProjectRo
                 ErrorUtilities.VerifyThrowInternalNull(rootElement, "projectRootElement");
                 ErrorUtilities.VerifyThrow(rootElement.FullPath.Equals(key, StringComparison.OrdinalIgnoreCase),
                     "Got project back with incorrect path");
+
+                AddEntry(rootElement);
```

Comment:
May I ask you to explain this statement? We're inside the `GetOrAdd` callback so `rootElement` will be added to the cache with the key of `projectFile`. Why are we calling another add?

Reply:
Yes, I removed AddEntry from the end of all the open functions in order to divide downloading and adding to the cache code. So, now everywhere else where I used open functions, I now need to do AddEntry exactly afterwards, so that behavior is exactly the same as before.
Here, in case of SimpleProjectRootElementCache, AddEntry does a bit more than just adding to _cache (it raises an event additionally).

Reply:
Thank you, yes, it does what it was doing before. It doesn't look right, though, to be adding inside the `GetOrAdd` callback. It means that the item will be added twice - by `AddEntry` and then attempted another add by `GetOrAdd` after the callback returns. It looks as though this should be a `TryGetValue` instead of `GetOrAdd`.

Just a perf nit, really, and OK to ignore.